### PR TITLE
Reference tracking reduction

### PR DESCRIPTION
**Pull request by myronkscott**

### CODE REVIEW

File: common/src/main/java/com/tc/async/api/SEDA.java [hunk 1]
```diff
@@ -48,25 +48,4 @@ public StageManager getStageManager() {
   protected TCThreadGroup getThreadGroup() {
     return this.threadGroup;
   }
-  
-  public void stageWarning(Object description) {
-    
-  }
-  
-  private static class StallWarning {
```
Comment (chrisdennis) [inline common/src/main/java/com/tc/async/api/SEDA.java:56]: Is this change unrelated?

File: common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java [hunk 2]
```diff
@@ -175,8 +177,11 @@ public CommunicationsManagerImpl(MessageMonitor monitor, TCMessageRouter message
     this.connectionManager = connMgr;
 
     Assert.eval(healthCheckerConfig != null);
+    ReferenceQueue<Object> gc = new ReferenceQueue<>();
+    PhantomReference<Object> ref = new PhantomReference<>(this, gc);
```
Comment (chrisdennis) [inline common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java:181]: From memory phantom references are more expensive during GC than weak ones. Not that there'll be many of them... but does this need to be phantom?

File: common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java [hunk 3]
```diff
@@ -175,8 +177,11 @@ public CommunicationsManagerImpl(MessageMonitor monitor, TCMessageRouter message
     this.connectionManager = connMgr;
 
     Assert.eval(healthCheckerConfig != null);
+    ReferenceQueue<Object> gc = new ReferenceQueue<>();
```
Comment (chrisdennis) [inline common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java:180]: This logic could go inside the if condition, right?

File: tc-client/src/main/java/com/tc/object/DistributedObjectClient.java [hunk 4]
```diff
@@ -473,13 +428,13 @@ public ClientEntityManager getEntityManager() {
...
   public String getClientState() {
     PrettyPrinter printer = new MapListPrettyPrint();
-    this.communicationStageManager.prettyPrint(printer);
     this.clientEntityManager.prettyPrint(printer);
     return printer.toString();
   }
 
   public void dump() {
     DSO_LOGGER.info(getClientState());
+    DSO_LOGGER.info(ThreadDumpUtil.getThreadDump());
```
Comment (chrisdennis) [inline tc-client/src/main/java/com/tc/object/DistributedObjectClient.java:437]: Was this just for debugging... or is it compensating for losing the `communicationsStageManager`?

**Review by chrisdennis — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/tc/async/api/SEDA.java (hunk 1 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/async/api/SEDA.java:56]: Cleanup of dead code

Comment (myronkscott) [reply common/src/main/java/com/tc/async/api/SEDA.java:56]: was holding references not needed

File: common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java (hunk 2 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java:181]: I was under the impression that phantom references where a replacement for finalization which this is basically an implementation of.

File: tc-client/src/main/java/com/tc/object/DistributedObjectClient.java (hunk 4 above)
Comment (myronkscott) [reply tc-client/src/main/java/com/tc/object/DistributedObjectClient.java:437]: It was doing a Thread.dumpStack previously which my IDE doesn't like.

**Review by chrisdennis — APPROVED** [review]

**Pull request merged**